File: IT/1/Containers/Esercizi/Java/Esercizio25.java
/* file Esercizio25.java */

import java.util.SortedSet;
import java.util.TreeSet;

public class Esercizio25 {
	public static void main(String [] args) {
		SortedSet<Integer> a = new TreeSet<Integer>();

		a.add(1);
		a.add(2);
		a.add(3);

		SortedSet<Integer> b = new TreeSet<Integer>();

		b.add(2);
		b.add(3);
		b.add(4);

		SortedSet<Integer> c = new TreeSet<Integer>();

		c.add(3);
		c.add(4);
		c.add(5);

		System.out.println("A = " + a);
		System.out.println("B = " + b);
		System.out.println("C = " + c);

		SortedSet<Integer> intersection = new TreeSet<Integer>(a);
		intersection.retainAll(b);
		intersection.retainAll(c);

		System.out.println("Intersezione tra gli insiemi A, B e C: " + intersection);
	}
}

File: IT/1/Containers/Esercizi/C++/esercizio25.cpp
/*
file esercizio24.cpp
compilare con: g++ -std=c++11 esercizio24.cpp
*/

#include <iostream>
#include <iterator>
#include <algorithm>
#include <set>

using std::cout;
using std::endl;
using std::ostream_iterator;
using std::set;

int main()
{
	set<int> a = {1, 2, 3};
	set<int> b = {2, 3, 4};
	set<int> c = {3, 4, 5};

	cout << "A = [";
	copy(a.begin(), a.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	cout << "B = [";
	copy(b.begin(), b.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	cout << "C = [";
	copy(c.begin(), c.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	set<int> temp;
	set_intersection(a.begin(), a.end(), b.begin(), b.end(), inserter(temp, temp.begin()));

	set<int> intersection;
	set_intersection(temp.begin(), temp.end(), c.begin(), c.end(), inserter(intersection, intersection.begin()));

	temp.clear();

	cout << "Intersezione tra gli insiemi A, B e C: [";
	copy(intersection.begin(), intersection.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;
}

